pkg/mimirtool/client/backfill.go

```diff
@@ -100,11 +101,20 @@ func (c *MimirClient) backfillBlock(ctx context.Context, blockDir string, logctx
 		}
 	}
 
-	resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
-	if err != nil {
-		return errors.Wrap(err, "request to finish block upload failed")
+	for {
+		resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
+		if err != nil {
+			if strings.Contains(err.Error(), "429 Too Many Requests") {
```

Comment:
Can we modify `checkResponse` (called from `c.doRequest`) to check for status code 429 and return an error that we can check for? We shouldn't be checking for status message string.